experimental/Schemes/ToricBlowups.jl

```diff
@@ -0,0 +1,67 @@
+@attributes mutable struct ToricBlowdownMorphism{
+                                                 DomainType <: NormalToricVariety, 
+                                                 CodomainType <: NormalToricVariety
+    } <: AbsSimpleBlowdownMorphism{DomainType,
+                                   CodomainType,
+                                   Nothing,
+                                   ToricBlowdownMorphism
+                                  }
+  toric_morphism::ToricMorphism
+  new_ray::AbstractVector{<:IntegerUnion}
+
+  # additional fields for caching
+  center::IdealSheaf
+  exceptional_divisor::CartierDivisor
+
+  function ToricBlowdownMorphism(
+      Y::NormalToricVariety, new_ray::AbstractVector{<:IntegerUnion}, 
+      coordinate_name::String, set_attributes::Bool
+    )
+    bl = _blow_up(Y, star_subdivision(Y, new_ray); 
+                  coordinate_name = coordinate_name, 
+                  set_attributes = set_attributes
+                 )
+    return new{typeof(domain(bl)), typeof(codomain(bl))}(bl, new_ray)
+  end
+end
+
+# The user facing constructor and candidate to replace the current method of 
+# `blow_up`, eventually. Called with the weird name for the time being.
+function wrapped_blow_up(v::NormalToricVarietyType, new_ray::AbstractVector{<:IntegerUnion}; coordinate_name::String = "e", set_attributes::Bool = true)
+  return ToricBlowdownMorphism(v, new_ray, coordinate_name, set_attributes)
+end
+
+# Forwarding all the essential functionality for morphisms of schemes 
+underlying_morphism(bl::ToricBlowdownMorphism) = bl.toric_morphism
```

Comment:
This could, for instance, be used to get back the familiar toric morphism. All the special dispatch we already implemented for that one is automatically forwarded.

Reply:
I like it!